### DISCUSSION TITLE
Publishing to Linux produce wrong output file type

### DISCUSSION THREAD

**Discussion by Zilkode** [AvaloniaUI/Avalonia#7966] [Q&A]
When publishing my project to Linux the output file is only executable from terminal (./executable).
I want my app to be executable from UI (like Ubuntu) with double-click.
This is the scenario happens both when publishing from Windows and Mac.

When viewing the file type it states "shared library (application/x-sharedlib)", while a native Linux file type is "application/x-executable".

I have read that when compiling CPP you should compile with -fno-pie and -no-pie flags to produce application/x-executable.
Is there a workaround for this?

Thanks.

**Comment by maxkatz6** [comment]
In Avalonia we do not handle applications publishing. Can you open issue at dotnet/runtime or dotnet/sdk repositories?
